syscalls.c

```diff
@@ -360,10 +360,17 @@ int syscalls_nsleep(void *ustack)
 
 	if (ret == -EINTR) {
 		proc_gettime(&stop, NULL);
-		diff = stop - start;
-		if (diff < us) {
-			*sec = diff / (1000 * 1000);
-			*nsec = (diff % (1000 * 1000)) * 1000;
+		elapsed = stop - start;
+		if (us > elapsed) {
+			unslept = us - elapsed;
+		}
+		else {
+			unslept = 0;
+		}
+
+		if (unslept > 0) {
```

Comment:
`us == elapsed` is very unlikely, nevertheless, this will behave well under this edge case
```suggestion
		elapsed = stop - start;
		if (us > elapsed) {
			unslept = us - elapsed;
```

Reply:
This code also handles cases where elapsed is greater than us, which, since time_t is unsigned, would cause unslept to become a very large number instead of 0.

Reply:
Doesn't matter, in this case we don't use value of unslept

Reply:
I see it now